modules/auxiliary/gather/rancher_authenticated_api_cred_exposure.rb

```diff
@@ -0,0 +1,191 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Rancher Authenticated API Credential Exposure',
+        'Description' => %q{
+          An issue was discovered in Rancher versions up to and including
+          2.5.15 and 2.6.6 where sensitive fields, like passwords, API keys
+          and Ranchers service account token (used to provision clusters),
+          were stored in plaintext directly on Kubernetes objects like Clusters,
+          for example cluster.management.cattle.io. Anyone with read access to
+          those objects in the Kubernetes API could retrieve the plaintext
+          version of those sensitive data.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'h00die', # msf module
+          'Florian Struck', # discovery
+          'Marco Stuurman' # discovery
+        ],
+        'References' => [
+          [ 'URL', 'https://github.com/advisories/GHSA-g7j7-h4q8-8w2f'],
+          [ 'URL', 'https://github.com/fe-ax/tf-cve-2021-36782'],
+          [ 'URL', 'https://fe.ax/cve-2021-36782/'],
+          [ 'CVE', '2021-36782']
+        ],
+        'DisclosureDate' => '2022-08-18',
+        'DefaultOptions' => {
+          'RPORT' => 443,
+          'SSL' => true
+        },
+        'Notes' => {
+          'Stability' => [],
+          'Reliability' => [],
+          'SideEffects' => []
+        }
+      )
+    )
+    register_options(
+      [
+        OptString.new('USERNAME', [ true, 'User to login with', '']),
+        OptString.new('PASSWORD', [ true, 'Password to login with', '']),
```

Comment:
```suggestion
        OptString.new('USERNAME', [ true, 'User to login with']),
        OptString.new('PASSWORD', [ true, 'Password to login with']),
```
No reason to prepopulate with invalid values.